### PR TITLE
Integ Tests for Awareness Attribute Decommissioning

### PR DESCRIPTION
**Pull request by imRishN**
### Description
This PR adds integration tests for awareness attribute decommission

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java [hunk 1]
```diff
@@ -56,6 +72,425 @@ public void cleanup() throws Exception {
         assertNoTimeout(client().admin().cluster().prepareHealth().get());
     }
```
Comment (Gaganjuneja) [inline server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java:74]: How can we verify that nodes are drained before initiating the decommissioning and once prepared for decommission no new request is being sent to these nodes?

File: server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java [hunk 2]
```diff
@@ -56,6 +72,425 @@ public void cleanup() throws Exception {
...
+
+    public void testNodesRemovedAfterZoneDecommission_ClusterManagerNotInToBeDecommissionedZone() throws Exception {
+        assertNodesRemovedAfterZoneDecommission(false);
+    }
+
+    public void testNodesRemovedAfterZoneDecommission_ClusterManagerInToBeDecommissionedZone() throws Exception {
+        assertNodesRemovedAfterZoneDecommission(true);
+    }
+
+    public void testInvariantsAndLogsOnDecommissionedNodes() throws Exception {
```
Comment (Gaganjuneja) [inline server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java:158]: Methods are very big around 150 lines in some cases. It's hard to read in a single go. Can we do something about it?

File: server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java (hunk 1 above)
Comment (imRishN) [reply server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java:74]: We can never certainly say if the node was drained completely or not. Although before initiating the decommission, we ensure weights for decommissioned zone is 0 but the metrics where we track HTTP requests doesn't get to zero. By the time we come up with an effective proposal for draining, we have implemented a 2 min wait after setting weights to 0 and executing decommission. 
Implemented as part of this PR - https://github.com/opensearch-project/OpenSearch/pull/4586

File: server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java (hunk 2 above)
Comment (imRishN) [reply server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java:158]: I tried to abstract out but use cases for each of these are unique and hence kept it like this

File: server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java [hunk 3]
```diff
@@ -56,6 +72,425 @@ public void cleanup() throws Exception {
...
+        DecommissionRequest decommissionRequest = new DecommissionRequest(decommissionAttribute);
+        assertBusy(() -> {
+            DecommissioningFailedException ex = expectThrows(
+                DecommissioningFailedException.class,
+                () -> client().execute(DecommissionAction.INSTANCE, decommissionRequest).actionGet()
+            );
+            assertTrue(ex.getMessage().contains("invalid awareness attribute requested for decommissioning"));
+        });
+    }
+
```
[the remark this answers is not in the document]
Comment (imRishN) [reply server/src/internalClusterTest/java/org/opensearch/cluster/coordination/AwarenessAttributeDecommissionIT.java:108]: Added a test for single zone decommission. Quorum loss checks is internally handled during reconfiguration of Voting Configuration

**Comment by imRishN** [comment]
@Bukhtawar Can you please also review this PR?
@Gaganjuneja Can you please it again?

**Review by Bukhtawar — APPROVED** [review]

**Pull request merged**